crates/harp/src/environment_iter.rs

```diff
@@ -13,20 +11,20 @@ pub struct EnvironmentIter {
     names: std::vec::IntoIter<String>,
 }
 
-#[derive(Eq, PartialEq)]
+#[derive(Eq)]
 pub struct Binding {
     pub name: RSymbol,
     pub value: BindingValue,
 }
 
-// What is this used for? Do we still need it?
-// https://github.com/posit-dev/amalthea/commit/04a9fe20a48cb01c4fe2d6fe1a4dfdc0f2a186fc
-#[derive(Eq)]
-pub struct BindingNestedEnvironment {
-    pub has_nested_environment: bool,
+// Bindings are equal if their names and values are exactly the same.
+impl PartialEq for Binding {
+    fn eq(&self, other: &Self) -> bool {
+        self.name == other.name && self.value == other.value
+    }
```

Comment:
Does this do something different than the automatic derivation? If not (it looks like it), probably best to keep deriving it to avoid suggesting something special is going on.